File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int it = 0;
        for (int is = 0; is < s.size(); is++) {
            if (s[is] == t[it]) {
                it++;
            }
        }
        return t.size() - it;
    }
};

File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int i = 0, j = 0, result = 0;
        while (i < s.length() && j < t.length()) {
            char ch1 = s.charAt(i), ch2 = t.charAt(j);
            if (ch1 == ch2) {   
                j += 1;
            }
            i += 1;
        }
        return t.length() - j;
    }
}